src/import/pull.c

```diff
@@ -119,14 +119,39 @@ static int pull_tar(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return log_error_errno(r, "Failed to allocate puller: %m");
 
-        r = tar_pull_start(pull, url, local, arg_force, arg_verify, arg_settings);
+        r = tar_pull_start(pull, url, local, arg_force, arg_verify, arg_settings, false);
         if (r < 0)
                 return log_error_errno(r, "Failed to pull image: %m");
 
         r = sd_event_loop(event);
         if (r < 0)
                 return log_error_errno(r, "Failed to run event loop: %m");
 
+        if (r == 404) {
+                /* restart verification jobs using SHA256SUMS files */
+                sd_event_unref(event);
+
+                r = sd_event_new(&event);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to get default event source: %m");
+
+                assert_se(sigprocmask_many(SIG_BLOCK, NULL, SIGTERM, SIGINT, -1) >= 0);
+                (void) sd_event_add_signal(event, NULL, SIGTERM, interrupt_signal_handler,  NULL);
+                (void) sd_event_add_signal(event, NULL, SIGINT, interrupt_signal_handler, NULL);
+
+                r = tar_pull_new(&pull, event, arg_image_root, on_tar_finished, event);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to allocate puller: %m");
+
+                r = tar_pull_start(pull, url, local, arg_force, arg_verify, arg_settings, true);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to pull checksum file: %m");
+
+                r = sd_event_loop(event);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to run event loop: %m");
+       }
```

Comment:
as mentioned: this fallback logic should be here, it should be inside of the pull logic, and not visible here.